src/services/auth.service.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  IUser,
+  IUserCreateSchema,
+  tempUserArr,
+} from '../interfaces/user.interface';
+import HttpException from '../application/utils/exceptions/http-exceptions';
+import HttpStatusCode from '../application/utils/exceptions/statusCode';
+import {
+  generateAccessToken,
+  generateRefreshToken,
+} from '../application/utils/tokenForAuth/generateToken';
+
+export async function register(
+  newUser: IUserCreateSchema,
+): Promise<IUserCreateSchema> {
+  const { username, password } = newUser;
+
+  if (!username || !password) {
+    throw new HttpException(
+      HttpStatusCode.BAD_REQUEST,
+      `Please provide username and password`,
+    );
+  }
+
+  if (tempUserArr.find((user) => user.username === username)) {
+    throw new HttpException(
+      HttpStatusCode.BAD_REQUEST,
+      `User with username ${username} already exists`,
+    );
+  }
+  tempUserArr.push(newUser);
+  return newUser;
+}
+
+export async function login(
+  credentials: IUserCreateSchema,
+): Promise<{ accessToken: string; refreshToken: string }> {
+  const { username, password } = credentials;
+
+  if (!username || !password) {
+    throw new HttpException(
+      HttpStatusCode.BAD_REQUEST,
+      `Please provide username and password`,
+    );
+  }
+  const user = tempUserArr.find(
```

Comment:
Could you please clarify how exactly will user.repository look, or it will be just blank file where for now will be array for users and then we will add database logic?

Reply:
It will be implemented in two my next pull requests, so i see it like:
* `repositories` folder
* `user.repository.ts` class with some basic CRUD methods (i will add them)

So in this example this class should have `getByUsername(username: string)` method when your temporary logic is implemented

Also WDYT about controller-service-repository levels of abstraction?

Reply:
If you use it, there can be method like `authenticate(username, password)` which will find user, compare its password with encrypted one and send tokens

Reply:
We could use it, I'm all in. For now, my suggestion is to wait your pull request and then i will change logic of auth service functions, which means:

1)Delete temporary array of users
2) Switch to db entities and for example in **login** function - 

```  
const user = tempUserArr.find(
    (u) => u.username === username && u.password === password,
  ); 
```  
There will be as you've said ```getByUsername(username: string)``` which will find user from db. Everything else will be the same for **login** function.

If you have some suggestions or corrections please don't hesitate to tell them :)